lib/os/fdtable.c

```diff
@@ -371,6 +371,8 @@ int zvfs_close(int fd)
 
 	zvfs_free_fd(fd);
 
+	(void)sock_obj_core_dealloc(fd);
```

Comment:
This is called by `zsock_close()` so wondering why we would need to call it again here. What kind of socket is this, a TLS one?

Reply:
Also, I don't think we should call socket specific API here in the generic code.

Reply:
This is a TCP socket.
sock_obj_core_dealloc is indeed called by zsock_close, however with POSIX_API enabled we never reach zsock_close, and comparing the two functions the only real difference is that zsock_close actually calls sock_obj_core_dealloc, which properly closes the socket.
zsock_close does the following: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/sockets/sockets.c#L137

Reply:
Yeah, I would agree - the vtable call should deallocate any fs-specific object associated with the generic fd.